DQ_test.py

```diff
@@ -24,8 +24,27 @@
 import numpy
 from dqrobotics import *
 from DQ_test_facilities import get_list_of_dq_from_mat
+from DQ_test_facilities import get_list_of_matrices_from_mat
 
-mat = scipy.io.loadmat('DQ_test.mat')
+"""
+Note from Juan Jose Quiroz Omana:
+
+I generated new data (DQ_test.mat) using DQ_test.m to include Q8 operations. 
+The legacy data (October 10, 2019) is in DQ_test_october_10_2019.mat.
+However, The test_ad fails with 'DQ_test.mat' with the following assertion:
+
+AssertionError: 
+Arrays are not almost equal to 12 decimals
+Error in Ad
+Mismatched elements: 1 / 8 (12.5%)
+Max absolute difference: 1.8189894e-12
+Max relative difference: 5.05591348e-16
+
+This is the reason to use 'DQ_test_october_10_2019.mat' for all tests except 
+test_of_Q4 and test_of_Q8, which use 'DQ_test.mat.
+"""
+mat = scipy.io.loadmat('DQ_test_october_10_2019.mat')
+mat_feb_22_2023 = scipy.io.loadmat('DQ_test.mat')
```

Comment:
@juanjqo Hopefully the prior comment will also fix this, but otherwise I think that the filename and the variable name should be compatible.

Reply:
@mmmarinho I agree that in cases where there are several mat files, their names should be compatible. However, since we decided to use only one mat file for all tests, renaming it could be inconvenient and maybe unnecessary (?).
For instance, most of the tests in DQ_test.py read the mat file. Therefore, (please correct me if I'm wrong) each time we generate a new mat file, we must rename it in each test. 
Example:
Current:  (with mat_06_04_2023)
```python
# Binary operators
    def test_plus(self):
        result_of_plus = get_list_of_dq_from_mat('result_of_plus', mat_06_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_plus):
            self.assertEqual(a + b, c, "Error in +")

    def test_minus(self):
        result_of_minus = get_list_of_dq_from_mat('result_of_minus', mat_06_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_minus):
            self.assertEqual(a - b, c, "Error in -")

    def test_times(self):
        result_of_times = get_list_of_dq_from_mat('result_of_times', mat_06_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_times):
            self.assertEqual(a * b, c, "Error in *")

    def test_dot(self):
        result_of_dot = get_list_of_dq_from_mat('result_of_dot', mat_06_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_dot):
            self.assertEqual(dot(a, b), c, "Error in dot")
```
Let say that we generate a new mat file (mat_21_04_2023), in that case we would need to modify each test with the new name right?

```python
# Binary operators
    def test_plus(self):
        result_of_plus = get_list_of_dq_from_mat('result_of_plus', mat_21_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_plus):
            self.assertEqual(a + b, c, "Error in +")

    def test_minus(self):
        result_of_minus = get_list_of_dq_from_mat('result_of_minus', mat_21_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_minus):
            self.assertEqual(a - b, c, "Error in -")

    def test_times(self):
        result_of_times = get_list_of_dq_from_mat('result_of_times', mat_21_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_times):
            self.assertEqual(a * b, c, "Error in *")

    def test_dot(self):
        result_of_dot = get_list_of_dq_from_mat('result_of_dot', mat_21_04_2023)
        for a, b, c in zip(dq_a_list, dq_b_list, result_of_dot):
            self.assertEqual(dot(a, b), c, "Error in dot")
```
I mean, it is not big deal, but it could be inconvenient. What do you think?

Best regards, 

Juancho

Reply:
@juanjqo I completely agree, in fact I think this is a perfect example of what I meant to say initially. 😊